pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -210,6 +210,7 @@ def save_checkpoint(self, trainer, pl_module):
             or self.period < 1  # no models are saved
             or (epoch + 1) % self.period  # skip epoch
             or trainer.running_sanity_check  # don't save anything during sanity check
+            or trainer.evaluating  # don't save anything during evaluation: might delete the checkpoint being evaluated
```

Comment:
You need to add a similar condition in the `EarlyStopping` callback too.

Reply:
Thanks, I had missed that. Added in the latest commit.